assignment3/GridWordSearcher.java

```diff
@@ -0,0 +1,121 @@
+package assignment3;
+
+import java.util.*;
+
+public class GridWordSearcher {
+    private int n;
+    private int m;
+    private char[][] grid;
+    private WordDictionary dict;
+    private static List<List<Integer>> directions;

```

Comment:
I think making this a List<Pair<Integer>> or even introducing a new class:

public class Step:
  int xStep;
  int yStep;
}

would be clearer to read. You can't tell what a List<List<Integer>> means just by looking at the type, but you can with a class List<Step> steps.